src/Generators/DataTablesMakeCommand.php

```diff
@@ -186,7 +186,7 @@ protected function getStub()
     protected function replaceFilename(&$stub)
     {
         $stub = str_replace(
-            'DummyFilename', str_slug($this->getNameInput()), $stub
+            'DummyFilename', preg_replace('#datatable$#i', '', $this->getNameInput()), $stub
```

Comment:
Can you please explain what this code do? Can't recall ^_^. Thanks!

Reply:
@yajra  You mean this `replaceFilename` method? It replaces `DummyFilename` placeholder in `datatables.stub`, with `datatables:make` command.

Currently the default generated filename for exporting data table is "URL safe slug case" i.e. lower case of the class/file name. I think it is not necessary to slug a class name, but using the original class name seems more readable for a exported/downloaded data file, see PR description for example : )

Reply:
You can try `datatables:make` command and see the `filename` method in the generated file.

Reply:
What does this code `#datatable$#i` do? Why do we need to preg_replace again please?

Reply:
Maybe I still do not understand what you mean 😳 
`preg_replace` is used to format the user input name, e.g. `UserDataTable` or `UserDatatable` to `User`, by removing the finishing `DataTable` string in the name.
`str_replace` is replacing the `DummyFilename` in the stub content to a default filename for the `filename()` method.

Actually I did not use the dataTable export buttons in my projects, maybe I made mistakes on something...

Reply:
Ok I think I got this now, you are just stripping the dataTable word in the filename for export. Sorry a bit slow to get it :).

Reply:
😛